### ISSUE TITLE
[BUG] Doubled app after installing

### ISSUE THREAD

**Issue by jesmrec** [owncloud/android#4262]
### Steps to reproduce

1. Install the app from scratch
2. Open the app

### Actual behaviour

App is doubled:

https://github.com/owncloud/android/assets/14894746/757ad1fd-1224-4d61-88d7-b57d23c69e27

only happens if there is no accounts

### Expected behaviour

app only once

Pixel 2, Android 11

Reproducible in v4.1.1 and so on

**Comment by Aitorbp** [comment]
We previously had a security issue, which we resolved by adding `askAffinity=“”` in the `Manifest`. Troubleshooting this issue also [suggested](https://medium.com/mobis3c/android-task-hijacking-6a3a8848f16e) that you set `launchMode` to `singleInstance` on the activity, which will prevent other activities from being part of your task. But this suggestion does not prevent the `SplashActivity` screen from appearing in `Recents Screen`, therefore, I rule it out as a solution, since in addition to adding more code, this vulnerability is only applicable for devices lower than Android 9. In versions 28 and later of the Android SDK (Android 9) contain the appropriate operating system patches to prevent this vulnerability.

if we set `taskAffinity` property of the application activities to ""(empty string) in the <activity> tag of the AndroidManifest.xml to force the activities to use a randomly generated task affinity, or set it at the <application> tag to enforce on all activities in the application. The affinity indicates which task an activity prefers to belong to, and now it is unknown due to this implementation, so we have to try to handle which activities are going to be shown in the `Recents Screen`.

So I would keep taskAffinity=“” in the Manifest, even though it may have effects on task management and how activities are organized in the task stack. Here we get to the point of the issue: due to the `askAffinity=“”`, the SplashActivity appears in the Recents screen.

To prevent this, you must [use](https://developer.android.com/guide/topics/manifest/activity-element#lmode) `android:excludeFromRecents` manually so that the system does not open activities in Recent automatically. Additionally this other flag `android:excludeFromRecents` is used to ensure that the task does not appear in recent applications, thus hiding the application from the attacker.

To give more context: Normally, we let the system define how our tasks and activities are represented in the Recents screen. We don't need to modify this behavior. However, your app can determine how and when activities appear in the Recents screen and this is what is happening right now. In our case, I don't see any behavior that directly handles the `SplashActivity`, which is the screen that remains open in Recent Apps. Following the [documentation](https://developer.android.com/guide/components/activities/recents), the system is doing it internally.

But this solution does not work when the application is installed from scratch. Since by removing the splashActivity from the Recent Apps, the one corresponding to the FileDisplayActivity is activated, but since the view is not visible because it overlaps with the LoginActivity, it remains in the Recent Apps with a white background. So we have to remove this task manually as well at this point.

**Comment by Aitorbp** [comment]
We are facing a problem with task management in the Recents Screen. By adding `android:excludeFromRecents="true"`
in the `SplashScreen` the behaviour of the application is not as expected. That is to say, if we are in the Recents Screen and we find a task from onwCloud and another from another application, when we open this second application, the ownCloud application will close from Recents. 
To avoid this we have tried different formulas, from some conventional ones playing with the flags to manually delete the tasks we don't want and leave only the `LoginActivity` task, but the application is still closed.
The ideal option, following the documentation, would be to leave `android:excludeFromRecents="true"` in the `Manifest` and when we make the intent to LoginActivity add the following tags: `FLAG_ACTIVITY_MULTIPLE_TASK` and `FLAG_ACTIVITY_RETAIN_IN_RECENTS`.
`FLAG_ACTIVITY_MULTIPLE_TASK`: If you set this flag, when creating a new document, the system will always create a new task with the target activity as root. This setting allows to open the same document in several tasks. 
`FLAG_ACTIVITY_RETAIN_IN_RECENTS`: If you want to retain a task in the Recent screen, even if the activity is already finished, pass this flag to the addFlags() method of the intent that starts the activity.

This method did not work.

Similarly, the `FLAG_ACTIVITY_NEW_DOCUMENT` tag was also used, but this did not work either.

The use of the `android:documentLaunchMode` and `android:launchMode` tags and its various properties has also been managed. We have tried to follow some logic of the app to set these tags, but they did not give the expected result either.

Documentation: https://developer.android.com/guide/components/activities/recents?hl=es-419

**Comment by JuancaG05** [comment]
Hi @hannesa2, we would like to know if you have any ideas regarding this issue? We were trying several stuff as you can see in previous comments, but we didn't get to a valid solution. Any suggestions are welcome 😄

**Comment by hannesa2** [comment]
Can you reproduce it ? I didn't investigated.

With this https://github.com/owncloud/android/blob/master/owncloudApp/src/main/AndroidManifest.xml#L82 you see an icon in launcher, and having more then one, you would see more.
This would be my first shot when I would investigate it.

**Comment by hannesa2** [comment]
Open the apk directly in Android Studio and look into the given manifest ....

**Comment by JuancaG05** [comment]
> Can you reproduce it ? I didn't investigated.

Hi @hannesa2, yes, we can reproduce it, just as you can see in the video of the first comment.

> With this https://github.com/owncloud/android/blob/master/owncloudApp/src/main/AndroidManifest.xml#L82 you see an icon in launcher, and having more then one, you would see more. This would be my first shot when I would investigate it.

It's not that we have several icons for the app, the title of the issue may be misleading. The problem here is that when the app is opened (with the single icon we have for it), we have like 2 instances of the app because 2 different activities appear in the Recents screen, and we expect just 1.

**Comment by jesmrec** [comment]
As clue, we think this one could be a solution: https://github.com/owncloud/android/issues/3391, but it has more issues to take in account (MDM regards).

**Comment by hannesa2** [comment]
I checkout out latest master and I only see one ownCloud.
<img width="413" alt="image" src="https://github.com/owncloud/android/assets/3314607/25e39e55-7ec2-4b85-b64b-26634b63c13c">

Sorry, it's a local issue on your side

**Comment by JuancaG05** [comment]
Hi @hannesa2, as I commented, there is only a single icon on our side as well, so that is NOT the problem. The problem can be seen in the Recents screen (pressing the square button from the device's bottom bar in case of Pixels, as shown in the video in https://github.com/owncloud/android/issues/4262#issue-2043089936), where there are 2 different elements for ownCloud, when there should be just 1.

**Comment by Aitorbp** [comment]
I have try to implement `SplashScreenApi` to see if the splash screen is excluded from recents. But It isn't changed the behaviour. Unfortunately, the splash screen is showing in the Recents Apps

**Comment by DeepDiver1975** [comment]
Confirmed, still present on current `master` — and the title is the source of the confusion. This is **not** duplicate launcher icons: the manifest declares exactly one MAIN+LAUNCHER activity (`SplashActivity`, AndroidManifest.xml). The bug is **multiple Recents/task-switcher entries** on a fresh install.

Current manifest state (master) shows none of the experimental fixes from the thread were merged — it's still the "without changes" baseline:
- `SplashActivity` / `FileDisplayActivity`: no `excludeFromRecents`, no `launchMode`.
- `LoginActivity`: `launchMode="singleTask"`.
- `application`: `taskAffinity=""`.

So on first launch all three (Splash → FileDisplay → Login) land as separate Recents tasks, vs. the single task expected.

Note for whoever picks this up: the @Aitorbp findings showed `android:excludeFromRecents="true"` on Splash (+ FileDisplay) fixes the Recents duplication but regresses task retention when switching apps; the `FLAG_ACTIVITY_MULTIPLE_TASK` / `RETAIN_IN_RECENTS` / `NEW_DOCUMENT` combinations didn't help. #3391 is the related single-task-management track and carries MDM implications, so a launch-flag/taskAffinity change needs verifying against branded/MDM builds before merging.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
